presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -759,6 +763,16 @@ public boolean isOrderByRedundant(OrderBy orderBy)
                         field -> ImmutableMap.of(toSchemaTableName(field.getOriginTable().get()), field.getOriginColumnName().get())));
     }
 
+    public boolean isStatementRewriteWithOptimizationEnabled()
+    {
+        return isStatementRewriteWithOptimizationEnabled;
+    }
+
+    public void setStatementRewrite()
```

Comment:
"set" is confusing, which means "set to true". disableStatementRewriteWithOptimization()/enableStatementRewriteWithOptimization() would be better. You might also want to add enableStatementRewriteWithOptimization().